### ISSUE TITLE
Export for Web Portal

### ISSUE THREAD

**Issue by norinespears** [specify/specify7#786]
Currently only Specify 6 has the ability to export data for use in the web portal but Specify 7 users also wish to take advantage of the web portal as the public face for their collection.

reported by: several, including Idaho

**Comment by norinespears** [comment]
Ben: since it now accepts CSV it should be quite a bit easier!!

**Comment by maxxxxxdlp** [comment]
@benanhalt can this be implemented similarly to the "Export to CSV" button in Query builder?

Should it be available for all records or only for CO and dependent tables?
Are arbitrary mappings allowed, or is there specific schema that is accepted by the web portal?

**Comment by benanhalt** [comment]
Yeah it would mostly reuse the export to CSV mechanism. The portal exports also contain other information that has to be dealt with to implement this feature, though.

I think the webportal assumes it is dealing with collection object data, but otherwise it is pretty flexible about the mappings, I think.

**Comment by maxxxxxdlp** [comment]
For future reference:
Specify 6 has a remote pref webportal.PROCESS_WEBLINKS that is related to this

**Comment by grantfitzsimmons** [comment]
As @benanhalt said, this should be relatively straightforward given enough time. I am happy to work with any developer looking to solve this issue.

> Yeah it would mostly reuse the export to CSV mechanism. The portal exports also contain other information that has to be dealt with to implement this feature, though.
> 
> I think the webportal assumes it is dealing with collection object data, but otherwise it is pretty flexible about the mappings, I think.

**Comment by grantfitzsimmons** [comment]
This should be automatic for web portals hosted on Specify Cloud. 
Currently, it takes a decent chunk of time and labor from us to update these instances.
